dhis-2/dhis-api/src/main/java/org/hisp/dhis/dashboard/Dashboard.java

```diff
@@ -52,6 +53,16 @@
     
     private List<DashboardItem> items = new ArrayList<>();
 
+    /**
+     * Whether we restrict the filter dimensions the user may select.
+     */
+    private Boolean filterDimensionsAreRestricted;
```

Comment:
I think this persisted property is redundant. An empty `allowedFilterDimensions` could indicate false.

```java
public boolean hasFilterDimensionRestriction()
{
    return allowedFilterDimensions != null && !allowedFilterDimensions.isEmpty();
}
```

Reply:
I had thought of that. If we allow a dashboard designer to restrict the filter dimensions, I think we should allow them to disallow all filter dimensions. Note that this would be allowed for in Joe Cooper's configuration U/I design. Initially I had thought that a null value could be used for no restrictions, an empty set could be used for allowing no dimensions, and a non-empty set for allowing only the specified dimensions.

I initially coded it this way. But I found in testing that if a dashboard already has an (empty or non-empty) set in the database, and the dimension is rewritten into the database with the json value missing for allowedFilterDimensions, then the (empty or non-empty) set remains unchanged in the database. In the absence of finding a practical way for the frontend to tell the backend that a null should be stored instead of an empty set, I thought we would need the extra boolean. Let me know if you see a better way to do this.